### PR TITLE
Refactor data

### LINKED ISSUE #3854: Migrate data and references to it
There are a lot of places where data is referenced in Jekyll-style ways, like featured site content in the blog post layout, etc.

Review references to data across all layouts and make sure it's rendering correctly.

```[tasklist]
### Acceptance criteria
- [x] Change all site.baseurls from regular links, following guidance in docs/development.md (breadcrumbs/images handled in another task)
- [x] Handle all `find_collection`s
- [x] All `site.data` references are assessed and corrected
- [x] All other `site.` references are assessed and corrected
- [x] Tag index page is fixed
- [ ] (Make own ticket) All `page.` references are assessed and corrected
- [ ] (Make own ticket) All `env.` references are assessed and corrected (should we be switching on envs in templates?)
- [x] All `assign`s are assessed and corrected
- [ ] (Make own ticket) Fix favicons (stored now in _includes/layouts/favicons.yml)
```

```[tasklist]
### Layouts
- [x] author
- [x] base
- [x] blog-page
- [x] data
- [ ] (handled in other tickets re: page variables) default
- [x] favicons
- [x] guides
- [x] page
- [x] post
- [x] primary
- [x] project-page
- [x] styled-container
- [x] styleguide
- [x] tag-index
- [x] tag-results
- [x] wide
```

```[tasklist]
### Partials
- [x] analytics
- [x] banner-cta
- [x] breadcrumb
- [x] card-project
- [x] card-with-image-guides
- [x] card-with-image
- [x] feature-image
- [x] featured-posts
- [x] footer
- [x] graphic-block
- [x] header
- [x] heading-link
- [x] hero
- [x] highlights
- [x] logo
- [x] menu
- [x] meta
- [x] meta_redirect
- [x] pagination-links
- [x] post-author
- [x] post-authors
- [x] post-date
- [x] post-debug
- [x] post-excerpt
- [x] post-preview
- [x] post-previews
- [x] post-tags
- [x] post-title
- [x] scripts
- [x] side-cta
- [x] sidenav
- [x] site-alert
- [x] skipnav
- [x] social-media
- [x] styles
- [x] subnav
- [x] tagline
- [x] testimonial
- [x] usa-banner
- [x] usa-identifier
```

### PR DESCRIPTION
**Pull request by beechnut**
# Pull request summary

Overall, this PR:

- Refactors data into clearer organization, and fixes data references in all templates
- Removes unused layouts, partials, and data
- Adds `tags` collection and updates references to tags data
- Changes nearly all links from Jekyll format (`{{ site.baseurl }}`) to 11ty format (`{{ "path" | url }}`). Images and some navigation links remain.
- Updates site metadata  
- Cleans up some code, formatting, logic

Closes #3854 and #3859

### CODE REVIEW

**Comment by cantsin** [comment]
We get some `Unexpected unnamed function` linter errors -- possible to disable this in the linter? (I think these are valid uses of anonymous functions)

**Review by cantsin — APPROVED** [review]
LGTM, it builds. would like to see the linting errors fixed but that's not strictly necessary.

File: _data/guides.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
-#"promoted" is being used for layout, and per the design, it should only be set to true for two guides at a time
+# "promoted" is being used for layout, and per the design,
+# it should only be set to true for two guides at a time.
+# TODO: Create a validation to make sure it's only two
```
[the remark this answers is not in the document]
Comment (beechnut) [reply _data/guides.yml:3]: Listed in #3864 :)

File: _includes/layouts/default.html [hunk 2]
```diff
@@ -1,15 +1,15 @@
----
-main: foo, bar, qaz
```
[the remark this answers is not in the document]
Comment (beechnut) [reply _includes/layouts/default.html:2]: You're very welcome.

File: config/collections.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+// smell: This is not the slugify library used by 11ty
+// However, that library requires ESM and this project is CommonJS.
+// The best thing to do would be to import the slugify filter, but I'm not quite
+// sure how to do that from this separate file — perhaps dependency injection
+// in .eleventy.js?
```
[the remark this answers is not in the document]
Comment (beechnut) [reply config/collections.js:5]: Done, added "look at all todos" to the epic

**Comment by beechnut** [comment]
@cantsin I added "ignore" directives to the lines causing the eslint warnings. I also added a task to [post-migration tasks](https://github.com/18F/18f.gsa.gov/issues/3864) to remind us to double-check/un-ignore these eventually.

**Pull request merged**